### DISCUSSION TITLE
Commands with Input Parameters

### DISCUSSION THREAD

**Discussion by sounak-1989** [edgexfoundry/community#118] [General]
Hey guys,

I have a device with multiple channels. There are different sensors connected to each channel. I am trying to design the device profile so that I can pass in the channel which I want to read. 

Is there a way where I can send a read command from core command with the Channel number passed as a JSON body? I dont see any option to access the JSON body through the parameters passed to HandleReadCommands().

Let me know if there is a better way to do it or am I getting the whole thing wrong.

Thanks

**Comment by cloudxxx8** [comment]
which device protocol you are talking about?
you can pass http query parameters to Get command, and they will be passed to HandleReadCommands().
